core/src/main/java/com/kin/ecosystem/core/data/order/OutgoingTransferCall.java

```diff
@@ -0,0 +1,139 @@
+                            new KinCallback<Order>() {
+                                @Override
+                                public void onResponse(Order order) {
+                                    scheduleTimeoutTimer();
+                                }
+
+                                @Override
+                                public void onFailure(KinEcosystemException exception) {
+                                    callback.onOutgoingTransferFailed(request, exception.getMessage());
+                                }
+                            });
+                }
+            });
+        } catch (OperationFailedException e) {
+            callback.onOutgoingTransferFailed(request, e.getMessage());
+        }
+    }
+
+    private void listenToPayment(final OpenOrder order, @NonNull String transaction) {
+        createPaymentObserver(order, blockchainSource.extractTransactionId(transaction));
+        blockchainSource.addPaymentObservable(paymentObserver);
+    }
+
+
+    private void createPaymentObserver(final OpenOrder order, final String transactionId) {
+        paymentObserver = new Observer<Payment>() {
+            @Override
+            public void onChanged(final Payment payment) {
+                if (isTransferPayment(payment)) {
+                    blockchainSource.removePaymentObserver(this);
+                    //Cancel SSE timeout task
+                    if (!isTimeoutTaskCanceled.getAndSet(true)) {
+                        sseTimeoutTimer.cancel();
+                    }
+
+                    if (payment.isSucceed()) {
+                        callback.onOutgoingTransferSuccess(request, order, transactionId);
+                    } else {
+                        callback.onOutgoingTransferFailed(request, payment.getException().getMessage());
+                    }
+
+                }
+            }
+        };
+    }
+
+    private void scheduleTimeoutTimer() {
+        sseTimeoutTimer.schedule(new TimerTask() {
+            @Override
+            public void run() {
+                // TimerTask runs on a BG thread of the timer.
+                if (!isTimeoutTaskCanceled.getAndSet(true)) {
+                    // Timeout should be fulfilled, remove payment observer and start server polling for order.
+                    blockchainSource.removePaymentObserver(paymentObserver);
+                    sseTimeoutTimer.cancel(); // Clear queue so it can be GC
+                }
+            }
+        }, SSE_TIMEOUT);
+    }
+
+    private boolean isTransferPayment(Payment payment) {
+        return request.getMemo().equals(payment.getOrderID());
+    }
+
+
+    interface OutgoingTransferCallback {
+
+        void onOutgoingTransferSuccess(OutgoingTransferRequest request, OpenOrder order, String transactionId);
+
+        void onOutgoingTransferFailed(OutgoingTransferRequest request, String errorMessage);
```

Comment:
why we are passing only error msg ?
Can you pass the whole exception (KinEcosystemException)?